### ISSUE TITLE
Select product from list to promote

### ISSUE THREAD

**Issue by hafizrahman** [woocommerce/woocommerce-ios#11585]
From the technical docs:

> [...] due to the nature of the selection for the campaign creation, we’ll need to make some updates:
>
> 1. Update the selector to allow single-selection.
> 2. Hide the “Clear selection” button when single-selection is used.
> 3. Update the logic of the selector to treat all products as simple products, which means to disable specific logic of variable products and product bundles.

**Comment by itsmeichigo** [comment]
Hi @hafizrahman - I was about to pick this task but saw that you already did while #11559 is still in progress. I'm good with picking another issue, just pinging to ask if you were handling both? It's best to focus on one task at a time so that we don't step on each other's toes 😅

**Comment by hafizrahman** [comment]
@itsmeichigo 

From #11559, the task that's not done is the Yosemite one. I can't proceed with it because it depends on a previous PR #11573 that is still in review, and last week I was still waiting for your reply (on https://github.com/woocommerce/woocommerce-ios/pull/11573#discussion_r1437929511)

Since last weekend I was working mostly alone, I had to pick something else to work on while waiting. In the meanwhile, Iast week I also reviewed the counterpart of this issue on Android https://github.com/woocommerce/woocommerce-android/pull/10490. From that experience I thought I had good context of what is needed for this task, so I chose to work on this.

My intention is to finish this today, then jump back to #11559.

**Comment by itsmeichigo** [comment]
Thanks for the clarification! Usually, on iOS we would create another PR based on a branch that is not yet merged to trunk. That way we can unblock ourselves while waiting for PR reviews. I did the same with my PRs for the backend tasks of target options. Let me know if that makes sense to you.

A nice thing on GH is that when your base branch is merged to trunk, the dependent branch will be automatically updated to target trunk. That's why the above process works nicely for us without waiting for PR reviews.

**Comment by hafizrahman** [comment]
> Let me know if that makes sense to you.

It does, I've done this workflow before. In the case above, my thinking was that the parameter for the fetch function `fetchForecastedImpressions()` needs to be finalized yet before it can be used in the Yosemite task. That's why I waited. 

But in hindsight I suppose I could just use the current function signature and update it later on.
